### PR TITLE
Frikadellendiplom

### LINKED ISSUE #1167: Abfrage zur IfsG Belehrung
Es soll möglich sein, dass Engel das Vorhandensein der Belehrung nach § 43 Abs. 1 Infektionsschutzgesetz (aka Frikadellendiplom, sowohl "klein" als auch "groß"), analog zu Führerscheinen, eintragen können.
Auch diese Funktionalität soll als Soft-Requirement mit entsprechenden Hinweismeldungen zu Führerscheinen und Liste in der Tabelle des Engeltypen ausgegeben werden.
Potentielle Überprüfungen / Enforcen des status sind an dieser Stelle out of scope.

### PR DESCRIPTION
**Pull request by xuwhite**
fixes #1167

### CODE REVIEW

File: includes/view/AngelTypes_view.php [hunk 1]
```diff
@@ -161,6 +173,12 @@ function AngelType_view_buttons(
             icon('person-vcard') . __('my driving license')
         );
     }
+    if ($angeltype->requires_ifsg_certificate) {
```
Comment (MyIgel) [inline includes/view/AngelTypes_view.php:176]: (same here)

File: includes/view/AngelTypes_view.php [hunk 2]
```diff
@@ -173,6 +191,10 @@ function AngelType_view_buttons(
             error(__('This angeltype requires a driver license. Please enter your driver license information!'));
         }
 
+        if ($angeltype->requires_ifsg_certificate && !$user->license->ifsg_certificate) {
```
Comment (MyIgel) [inline includes/view/AngelTypes_view.php:194]: && here

File: src/Controllers/SettingsController.php [hunk 3]
```diff
@@ -222,6 +222,45 @@ public function saveLanguage(Request $request): Response
...
+
+    public function saveIfsgCertificate(Request $request): Response
+    {
+        $user = $this->auth->user();
+        $data = $this->validate($request, [
+            'ifsg_certificate' => 'optional|checked',
+            'ifsg_certificate_full' => 'optional|checked',
+        ]);
+
+        if (!config('ifsg_enabled')) {
```
Comment (MyIgel) [inline src/Controllers/SettingsController.php:251]: Eventually that check could be moved up a bit to not run more code than necessary but its fine as it is

**Review by MyIgel — CHANGES_REQUESTED** [review]

File: includes/view/AngelTypes_view.php [hunk 4]
```diff
@@ -111,6 +111,18 @@ function AngelType_edit_view(AngelType $angeltype, bool $supporter_mode)
                     __('Requires driver license'),
                     $angeltype->requires_driver_license
                 ),
+            $supporter_mode && config('ifsg_enabled') ?
```
[the remark this answers is not in the document]
Comment (xuwhite) [reply includes/view/AngelTypes_view.php:114]: Du bist dumm

[the remark this answers is not in the document]
Comment (MyIgel) [reply includes/view/AngelTypes_view.php:114]: Details!

File: includes/controller/user_driver_licenses_controller.php [hunk 5]
```diff
@@ -30,6 +30,28 @@ function user_driver_license_required_hint()
     return null;
 }
 
+function user_ifsg_certificate_required_hint()
+{
+    $user = auth()->user();
+
+    // User has already entered data, no hint needed.
+    if ($user->license->ifsg) {
```
[the remark this answers is not in the document]
Comment (xuwhite) [reply includes/controller/user_driver_licenses_controller.php:38]: done

File: includes/view/AngelTypes_view.php (hunk 1 above)
Comment (xuwhite) [reply includes/view/AngelTypes_view.php:176]: done

File: includes/view/AngelTypes_view.php (hunk 2 above)
Comment (xuwhite) [reply includes/view/AngelTypes_view.php]: done

File: src/Controllers/SettingsController.php (hunk 3 above)
Comment (xuwhite) [reply src/Controllers/SettingsController.php:251]: won't fix

**Pull request merged**